### DISCUSSION TITLE
Does it work with Raspberry Pi Zero?

### DISCUSSION THREAD

**Discussion by jackoGT** [outdoorbits/little-backup-box#191] [General]
Hi, I have a Raspberry Pi Zero W. 
Is it possible to install this project on this board?

**Comment by outdoorbits** [comment]
It might not be the fastest solution, but yes, it works.

Message ID: 
***@***.***>

**Comment by outdoorbits** [comment]
Did you try it? I would be very interested in transfer-rates. Maybe you could post some information?

What are the rates when you backup from
- external storage to another external storage?
- external storage to internal storage?

Did you convert images for view? Is the speed acceptable for you, whats the average time per image and what type/size of images did you convert?

**Comment by jackoGT** [comment]
With the old version, the backup speed was quite good. 
However, after installing the new version, the CPU usage reaches 100%, and the backup process becomes even slower. 
Anyway, I prefer using backup on an external memory device.

Can I ask you a question? 
Have you considered developing a lite version of the software that can be fully utilized on a Raspberry Pi Zero? 
The Pi Zero's small and compact size makes it an ideal device for backing up photos while traveling. It would be a great addition to the software's capabilities, and could potentially open up new opportunities for users who prefer to use smaller devices while on the go.

**Comment by outdoorbits** [comment]
The CPU load is 100% while no backup is running? Could you please login on a terminal and tell me, what 
> top
Gives back?

**Reply by jackoGT** [reply]
While it was not doing any backup and I was trying to set the settings, it was telling me that the CPU was at 100%. I couldn't even easily navigate from one menu item to another, it was very slow.

**Comment by outdoorbits** [comment]
There hasn't been any change to explain this. Is it possible to reproduce the issue? What have you done before?

**Reply by jackoGT** [reply]
Nothing. I installed the scripts, set the comitup wifi and then tried to set a few other small things like language, thumbnail, etc.
I had installed on the raspberry pi zero raspian lite 32-bit version.

**Comment by outdoorbits** [comment]
Could you please start top to see which process is responsible?

**Comment by rkosick** [comment]
I also used Raspberry Pi Zero long time ago, before I switched to Raspberry Pi Zero 2W. It worked fine back then. Could it be that the new WPA2 encryption of comitup is responsible? Can't imagine that this would create to much overhead for the Raspberry Pi Zero, but I don't know.
I could check it next week since I still have the old Raspberry Pi Zero.

**Reply by rkosick** [reply]
I saw that I had the Raspberry Pi Zero without the wireless chip. So I can't check if this is the problem. This long time ago I used it offline.

**Comment by outdoorbits** [comment]
I just installed lbb on pi zero and unfortunately I have to confirm your observation. But having a look at the output of "top", there is not one process taking all the power but its distributed to many processes. In this moment there was no active backup-task. It looks like the most power is taken by the actual raspberry pi os itself. Maybe you try to install lbb on an older os?

**Reply by jackoGT** [reply]
I will try an old version of raspian and see what happens. I'll tell you as soon as I have everything installed.

**Comment by jackoGT** [comment]
I tried installing lbb using 32-bit buster lite and the problem occurs again.

**Comment by outdoorbits** [comment]
Compared to which version (date of installation) did you recognize the 
changes in CPU-load?

**Reply by jackoGT** [reply]
I installed lbb for the first time in January.

**Comment by outdoorbits** [comment]
Does it make a difference to switch off the display? (Settings - Display - Display behavior:  Is a display connected and should it be used?)

**Reply by jackoGT** [reply]
No, I don't use the display and control the raspberry pi zero via web page.
In fact I disable all options concerning display and thumbnails (trying to save power).
I also tried disabling wifi and bluetooth from the terminal (controlling the pi via ethernet with an adapter) but nothing changed.

**Comment by outdoorbits** [comment]
Did you try to find out, which process is responsible for all the CPU load? Did you configure a standard backup method?

**Reply by jackoGT** [reply]
Using the "top" command, it shows me that the "udevadm trigger" process is the most heavily used one (about 43% of the cpu).
While running the "top" command, I was not performing any backups. Everything was on stand-by.
My standard backup method is from usb to usb.

**Comment by outdoorbits** [comment]
Could you please disable a standard backup, reboot and check again? 
Maybe checking for new devices explains the high CPU usage?

**Reply by outdoorbits** [reply]
I just installed a zero again and I can confirm, the CPU load is hight while lbb waits for a device. When backup is started, lbb checks every second for the expected device until it's found. I think, the best work around is too decide what's more important to you: 

1. Automatic starting backup but very slow GUI
2. GUI with acceptable reaction times but you need to start your backup manually.

**Reply by jackoGT** [reply]
Ok. However, without a default option it is a bit difficult to back up on the move, without the help of a wifi network.
Is it possible to use lbb via web-gui without being connected to a wifi network? It would be nice to maybe use it by using pi as a hotspot and control it via web-gui.

**Comment by outdoorbits** [comment]
Of course, that's what comitup does at first. Connect to the hotspot and navigate to https://10.41.0.1
If you don't want to use the GUI, just use a default backup method.

**Reply by jackoGT** [reply]
But when the initial screen for choosing wifi connections appears and clicking the "locate" button or not filling in the "hidden or missing SSID" part and clicking "submit", nothing happens.
This makes it necessary to use lbb only with a wifi network.

**Reply by rkosick** [reply]
You can hit the 3 dots on the upper right corner and go to: Just choose the WiFi as it is. Then you will stay connected with your phone, even if it is not providing any internet.

**Comment by outdoorbits** [comment]
Please have a look at the readme:

*** 2. Alternatively can reach the web UI of Little Backup Box when you 
are connected to its
*** hotspot as follows:
***
*** https://10.41.0.1 (secure, certificate can't be verified 
automatically, please confirm it)
*** http://10.41.0.1:8000 (insecure)
***

**Reply by jackoGT** [reply]
I apologize, it was my mistake. I was looking for something that would allow me to access the Web GUI from the initial configuration page, or I mistakenly entered 10.41.0.1 in the browser. However, when I entered the address correctly, everything worked perfectly. The backup time is quite acceptable.

**Reply by outdoorbits** [reply]
But the backup speed still became worse?

**Reply by jackoGT** [reply]
No, compared to before it is much better.
I made some test backups and it completed them in an acceptable time for me.
For 5Gb of photos it took approximately 20 min to make the backup.

**Reply by jackoGT** [reply]
I am really satisfied with the performance of Lbb on the Pi Zero.
However, is it also possible to install Lbb on other distributions such as Dietpi rather than Raspbian?

**Comment by outdoorbits** [comment]
I never tried. The install routine depends on apt. If all the parcels are available, maybe...?
